spring-batch-bigquery/src/main/java/org/springframework/batch/extensions/item/google/bigquery/BigQueryItemWriter.java

```diff
@@ -0,0 +1,437 @@
+                }
+            }
+        }
+        else {
+            Assert.isNull(this.writeChannelConfig.getSchema(), "Avro does not require schema");
+            Assert.isNull(this.writeChannelConfig.getAutodetect(), "Avro does not require autodetection");
+        }
+
+        Assert.notNull(this.writeChannelConfig.getFormat(), "Data format must be provided");
+
+        if (isCsv() || isJson()) {
+            Assert.notNull(this.rowMapper, "Row mapper must be provided");
+        }
+
+        if (Objects.nonNull(this.datasetInfo)) {
+            Assert.isTrue(
+                    Objects.equals(this.datasetInfo.getDatasetId().getDataset(), this.writeChannelConfig.getDestinationTable().getDataset()),
+                    "Dataset should be configured properly"
+            );
+        }
+
+        createDataset();
+    }
+
+    private boolean isCsv() {
+        return FormatOptions.csv().getType().equals(this.writeChannelConfig.getFormat());
+    }
+
+    private boolean isJson() {
+        return FormatOptions.json().getType().equals(this.writeChannelConfig.getFormat());
+    }
+
+    private boolean isAvro() {
+        return FormatOptions.avro().getType().equals(this.writeChannelConfig.getFormat());
+    }
+
+    private boolean isParquet() {
+        return FormatOptions.parquet().getType().equals(this.writeChannelConfig.getFormat());
+    }
+
+    private boolean isOrc() {
+        return FormatOptions.orc().getType().equals(this.writeChannelConfig.getFormat());
+    }
+
+    private boolean isBigtable() {
+        return FormatOptions.bigtable().getType().equals(this.writeChannelConfig.getFormat());
+    }
+
+    private boolean isGoogleSheets() {
+        return FormatOptions.googleSheets().getType().equals(this.writeChannelConfig.getFormat());
+    }
+
+    private boolean isDatastore() {
+        return FormatOptions.datastoreBackup().getType().equals(this.writeChannelConfig.getFormat());
+    }
+
+    private boolean tableHasDefinedSchema(Table table) {
+        return Optional
+                .ofNullable(table)
+                .map(Table::getDefinition)
+                .map(TableDefinition.class::cast)
+                .map(TableDefinition::getSchema)
+                .isPresent();
+    }
+
+    private void createDataset() {
+        if (Objects.nonNull(this.datasetInfo)) {
+            Dataset foundDataset = this.bigQuery.getDataset(this.writeChannelConfig.getDestinationTable().getDataset());
+
+            if (Objects.isNull(foundDataset)) {
+                this.bigQuery.create(this.datasetInfo);
+            }
+        }
+    }
+
+    /**
+     * It is expected that for {@link BigQueryItemWriter#isParquet()} and {@link BigQueryItemWriter#isOrc()} you use
+     * {@link Path} because there is no convenient way to write records to {@link OutputStream}.
+     * @deprecated not supported right now
+     */
+    @Deprecated
```

Comment:
API deprecation is typically used for public APIs. This is a private method.  It does not feel right to me to start v0.1.0 with something `@Deprecated`, and more importantly to use `@Deprecated` to denote something as not supported yet. I believe `UnsupportedOperationException` is more appropriate. If you do not want to throw an exception, just remove the method if it is not used, or remove this annotation.

Reply:
Done